utils/PHPStan/CheckFrameworkExceptionInstantiationViaStaticCallRule.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Utils\PHPStan;
+
+use CodeIgniter\Exceptions\FrameworkException;
+use PhpParser\Node;
+use PhpParser\Node\Expr\New_;
+use PhpParser\Node\Name\FullyQualified;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+
+final class CheckFrameworkExceptionInstantiationViaStaticCallRule implements Rule
```

Comment:
I think it should be named "CheckFrameworkExceptionInstantiationViaNamedConstructorRule": 

```suggestion
final class CheckFrameworkExceptionInstantiationViaNamedConstructorRule implements Rule
```